src/libraries/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -34,22 +35,35 @@ public abstract partial class FileSystemWatcherTest : FileCleanupTestBase
         /// Watches the Changed WatcherChangeType and unblocks the returned AutoResetEvent when a
         /// Changed event is thrown by the watcher.
         /// </summary>
-        public static (AutoResetEvent EventOccured, FileSystemEventHandler Handler) WatchChanged(FileSystemWatcher watcher, string[] expectedPaths = null)
+        public static (AutoResetEvent EventOccured, FileSystemEventHandler Handler) WatchChanged(FileSystemWatcher watcher, string[] expectedPaths = null, ITestOutputHelper _output = null)
         {
             AutoResetEvent eventOccurred = new AutoResetEvent(false);
 
-            FileSystemEventHandler changeHandler = (o, e) =>
+            FileSystemEventHandler handler = (o, e) =>
             {
-                Assert.Equal(WatcherChangeTypes.Changed, e.ChangeType);
+                if (e.ChangeType != WatcherChangeTypes.Changed)
+                {
+                    _output?.WriteLine("Unexpected event {0} while waiting for {1}", e.ChangeType, WatcherChangeTypes.Changed);
+                    Assert.Equal(WatcherChangeTypes.Changed, e.ChangeType);
+                }
+
                 if (expectedPaths != null)
                 {
-                    Assert.Contains(Path.GetFullPath(e.FullPath), expectedPaths);
+                    try
+                    {
+                        Assert.Contains(Path.GetFullPath(e.FullPath), expectedPaths);
+                    }
+                    catch (Exception ex)
+                    {
+                        _output?.WriteLine(ex.ToString());
+                        throw;
+                    }
```

Comment:
Why is all of this stuff changing?

Reply:
I've reverted the vast majority of the changes to this file.